internal/policy/verify.go

```diff
@@ -83,6 +83,186 @@ func VerifyRefFull(ctx context.Context, repo *gitinterface.Repository, target st
 }
 
+func VerifyMergeable(ctx context.Context, repo *gitinterface.Repository, targetRef, featureRef string) (bool, error) {
+	slog.Debug("Loading policy...")
+	policyState, err := LoadCurrentState(ctx, repo, PolicyRef)
+	if err != nil {
+		return false, err
+	}
+
+	slog.Debug("Loading current set of attestations...")
+	attestationsState, err := attestations.LoadCurrentAttestations(repo)
+	if err != nil {
+		return false, err
+	}
+
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", targetRef))
+	targetEntry, _, err := rsl.GetLatestUnskippedReferenceEntryForRef(repo, targetRef)
+	if err != nil {
+		return false, err
+	}
+
+	slog.Debug(fmt.Sprintf("Identifying latest RSL entry for '%s'...", featureRef))
+	featureEntry, _, err := rsl.GetLatestUnskippedReferenceEntryForRef(repo, featureRef)
+	if err != nil {
+		return false, err
+	}
+
+	mergeTreeID, err := repo.GetMergeTree(targetEntry.TargetID, featureEntry.TargetID)
+	if err != nil {
+		return false, err
+	}
+
+	authorization, githubApproval, err := getAttestationsForIndex(repo, attestationsState, targetRef, targetEntry.TargetID, mergeTreeID)
+	if err != nil {
+		return false, err
+	}
+
+	var (
+		rslEntrySignatureNeededForThreshold = false
+
+		gitNamespaceVerified  = false
+		pathNamespaceVerified = true // Assume paths are verified until we find out otherwise
+	)
+
+	verifiers, err := policyState.FindVerifiersForPath(fmt.Sprintf("%s:%s", gitReferenceRuleScheme, targetRef))
+	if err != nil {
+		return false, err
+	}
+	if len(verifiers) == 0 {
+		gitNamespaceVerified = true
+	}
+
+	for _, verifier := range verifiers {
+		// Try verifying twice
+		// Once with threshold as it is
+		// Once with threshold-- (to account for signature on gitObj)
+		err := verifier.Verify(ctx, gitinterface.ZeroHash, authorization, githubApproval)
+		if err == nil {
+			// we've reached threshold with just the approvals
+			gitNamespaceVerified = true
+			break
+		} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
+			// Unexpected error
+			return false, err
+		}
+
+		// If threshold == 1 and we have approval without gitObj signature, we'll have exited above
+		// Only try threshold-- if threshold > 1
+		if verifier.Threshold() > 1 {
+			// Make a copy before mutating verifier
+			verifier := verifier
+			verifier.threshold--
+
+			err := verifier.Verify(ctx, gitinterface.ZeroHash, authorization, githubApproval)
+			if err == nil {
+				rslEntrySignatureNeededForThreshold = true
+				gitNamespaceVerified = true
+				break
+			} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
+				return false, err
+			}
+		}
+	}
+
+	if !gitNamespaceVerified {
+		return false, fmt.Errorf("not enough approvals to meet Git namespace policies, %w", ErrUnauthorizedSignature)
+	}
+
+	hasFileRule, err := policyState.hasFileRule()
+	if err != nil {
+		return false, err
+	}
+	if !hasFileRule {
+		return rslEntrySignatureNeededForThreshold, nil
```

Comment:
Also not clicking what this has to do with file rules. If this is for mergeability, e.g. "the commit has the number of required RSL entries needed to meet the threshold", then this should be renamed since it reads like "RSL entries are needed to meet the threshold"